backend/controller/dal/testdata/go/fsmnext/fsmnext.go

```diff
@@ -0,0 +1,121 @@
+package fsmnext
+
+import (
+	"context"
+	"errors"
+
+	"github.com/TBD54566975/ftl/go-runtime/ftl"
+)
+
+func fsm() *ftl.FSMHandle {
+	// This FSM allows transitions moving forward through the alphabet
+	// Each transition also declares the next state(s) to transition to using State
+	//
+	//ftl:retry 2 1s
+	var fsm = ftl.FSM("fsm",
+		ftl.Start(StateA),
+		ftl.Transition(StateA, StateB),
+		ftl.Transition(StateA, StateC),
+		ftl.Transition(StateA, StateD),
+		ftl.Transition(StateB, StateC),
+		ftl.Transition(StateB, StateD),
+		ftl.Transition(StateC, StateD),
+	)
+	return fsm
+}
+
+type State string
+
+const (
+	A State = "A"
+	B State = "B"
+	C State = "C"
+	D State = "D"
+)
+
+type Event struct {
+	Instance     string
+	NextStates   []State            // will schedule fsm.Next with these states progressively
+	NextAttempts ftl.Option[int]    // will call fsm.Next this many times. Once otherwise
+	Error        ftl.Option[string] // if present, returns this error after calling fsm.Next() as needed
+}
+
+//ftl:typealias
+type EventA Event
+
+//ftl:verb
+func StateA(ctx context.Context, in EventA) error {
+	return handleEvent(ctx, Event(in))
+}
+
+//ftl:typealias
+type EventB Event
```

Comment:
What's the purpose of the type aliases? Unless there's a reason it would probably be simpler and less confusing to use `type EventB struct {}`